### ISSUE TITLE
Notations on transitions

### ISSUE THREAD

**Issue by choisy** [thinhong/denim#24]
I'm confused again about the first model in the README. In particular, if I understand well, when you use `mathexpr()`, the distribution that is implicitly assumed is an exponential one right? In that case, that would be much nicer if we could write
```
"V -> IV" = exponential(beta * 0.3 * V * (I + IV) / N)
```
instead of
```
"V -> IV" = mathexpr(beta * 0.3 * V * (I + IV) / N)
```
And, again, I don't think we need `mathexpr()` and `constant()` in this package.

**Comment by thinhong** [comment]
The reason I made the mathexpr in C++ was because I needed something to calculate with the number of people in >1 compartments (in this example we have 3 compartments V, I, IV), and if we want something more than just addition, substraction, multiplication... You're right in R we don't need a function for it, I can check if the object is a string then make it a mathexpr in C++.

What I understand from the exponential distribution is when we have a constant rate, i.e when the number going out of V is only `beta * V` then it is exponential, could you confirm? Anyway, in C++ it has to be handled by a mathexpr because there are >1 compartment and not only the left compartment involved lol.

Actually, I'm thinking about dropping the `exponential()` because when I set the `probability()` it will be the same as the `exponential()` as this is discrete time?

**Comment by thinhong** [comment]
I meant because the I and IV keep changing through time so beta * (I + IV) is not constant through time, and the length of stay in V is not exponential?

**Comment by choisy** [comment]
> I meant because the I and IV keep changing through time so beta * (I + IV) is not constant through time, and the length of stay in V is not exponential?

Yes, good point, you're right on this one. But please get rid off `mathexp()`, it's ugly. It's not because you need it in the C++ code that it should appear in the R interface. The whole point of the R interface is to hide as much as possible the technicalities of the C++ code. Thus, we want this user interface language as slim as possible.

**Comment by choisy** [comment]
Not sure what you mean about `probability()` here. Are you talking about `transprob()` by any chance? If yes, I think that's another good point on which you're right again, in which case the diagramme would now look like this right?
<img width="752" alt="Screen Shot 2022-04-06 at 17 57 32" src="https://user-images.githubusercontent.com/12168614/161960249-f910218b-18b0-4fa0-b30d-2d5f7b988122.png">
